### DISCUSSION TITLE
remapping Ctrl+D/U/F/B for use in insert mode.

### DISCUSSION THREAD

**Discussion by sourcevault** [NeoVintageous/NeoVintageous#928] [Q&A]
I am trying to make the vim Ctrl navigation keys to work in insert mode.

I do not have a particular question except to get feedback to know if I am doing things in the correct manner.

https://github.com/NeoVintageous/NeoVintageous/assets/20347647/efe46338-9d23-4c98-b7dd-234cec35b583

```vimrc
" .neovintageousrc

inoremap <C-d> :NvViCtrlD count=0 mode=mode_normal<CR>
inoremap <C-u> :NvViCtrlU count=0 mode=mode_normal<CR>
inoremap <C-b> :NvViCtrlB count=1 mode=mode_normal<CR>
inoremap <C-f> :NvViCtrlF count=1 mode=mode_normal<CR>

```

```.sublime-keymap
// .sublime-keymap

  {"keys": ["ctrl+d"],  "command": "nv_feed_key", "args": {"key": "<C-d>"},   "context": [{"key": "vi_insert_mode_aware"},  {"key": "nv_handle_key", "operand": "<C-d>"}]},
  {"keys": ["ctrl+u"],  "command": "nv_feed_key", "args": {"key": "<C-u>"},   "context": [{"key": "vi_insert_mode_aware"},  {"key": "nv_handle_key", "operand": "<C-u>"}]},
  {"keys": ["ctrl+b"],  "command": "nv_feed_key", "args": {"key": "<C-b>"},   "context": [{"key": "vi_insert_mode_aware"},  {"key": "nv_handle_key", "operand": "<C-b>"}]},
  {"keys": ["ctrl+f"],  "command": "nv_feed_key", "args": {"key": "<C-f>"},   "context": [{"key": "vi_insert_mode_aware"},  {"key": "nv_handle_key", "operand": "<C-f>"}]},
```

**Comment by gerardroche** [comment]
Yes, that will work. It might break in future updates because commands ike `nv_vi_ctrl_d` are not meant to be public apis.

I'm wondering if this will work:

```
inoremap <C-d> <Esc><C-d>
```

**Comment by gerardroche** [comment]
Yes it does. So you could actually re-enter insert mode:

```
inoremap <C-d> <Esc><C-d>i
```

**Reply by sourcevault** [reply]
![smart](https://github.com/NeoVintageous/NeoVintageous/assets/20347647/9da9f176-3072-49c5-b587-452cf2c265ad)

**Comment by gerardroche** [comment]
I've created an experimental package with the intent to provide insert mode keymaps https://github.com/gerardroche/NeoVintageousInsertModeKeymaps

I'm working on a review of how the keymaps are handled so a common package will be easier to update. 

p.s. I think I've found a way to support any character inputs and be able to use the in mappings too e.g. →, ³, ½, etc.
